core/templates/pages/contributor-dashboard-page/services/contribution-and-review-backend-api.service.ts

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend api service for fetching and resolving suggestions.
+ */
+
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { FetchSuggestionsResponse } from './contribution-and-review.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+interface ResolveToExplorationData {
+  action: string;
+  'review_message': string;
+  'commit_message': string;
+}
+
+interface ResolveToSkillData {
+  action: string;
+  'review_message': string;
+  'skill_difficulty': string;
+}
+
+interface UpdateTranslationData {
+  'translation_html': string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewBackendApiService {
+  private _SUBMITTED_SUGGESTION_LIST_HANDLER_URL = (
+    '/getsubmittedsuggestions/<target_type>/<suggestion_type>');
+  private _REVIEWABLE_SUGGESTIONS_HANDLER_URL = (
+    '/getreviewablesuggestions/<target_type>/<suggestion_type>');
+  private _SUGGESTION_TO_EXPLORATION_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/exploration/<exp_id>/<suggestion_id>');
+  private _SUGGESTION_TO_SKILL_ACTION_HANDLER_URL = (
+    '/suggestionactionhandler/skill/<skill_id>/<suggestion_id>');
+  private _UPDATE_TRANSLATION_HANDLER_URL = (
+    '/updatetranslationsuggestionhandler/<suggestion_id>');
+  private _UPDATE_QUESTION_HANDLER_URL = (
+    '/updatequestionsuggestionhandler/<suggestion_id>');
+
+  private _SUBMITTED_QUESTION_SUGGESTIONS = (
+    'SUBMITTED_QUESTION_SUGGESTIONS');
+  private _REVIEWABLE_QUESTION_SUGGESTIONS = (
+    'REVIEWABLE_QUESTION_SUGGESTIONS');
+  private _SUBMITTED_TRANSLATION_SUGGESTIONS = (
+    'SUBMITTED_TRANSLATION_SUGGESTIONS');
+  private _REVIEWABLE_TRANSLATION_SUGGESTIONS = (
+    'REVIEWABLE_TRANSLATION_SUGGESTIONS');
```

Comment:
These are to specify what kind of suggestions we want to fetch from the backend. We need this because urls were moved to the backend-api-service file. And in order to call the correct fetch function in the backend-api-service (fetchSubmittedSuggestionsAsync and fetchReviewableSuggestionsAsync), we need these variables.